### PR TITLE
fix: set optional max_instances to default value for CI

### PR DESCRIPTION
**Pull request by apeabody**
workaround for https://github.com/hashicorp/terraform-provider-google/issues/13263

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate .
╷
│
 
Warning: 
Deprecated
│
 
│
 
  with data.null_data_source.wait_for_files,
│
 
  on main.tf line 42, in data "null_data_source" "wait_for_files":
│
 
  42: data "null_data_source" "wait_for_files" 
{
│
 
│
 
The null_data_source was historically used to construct intermediate values
│
 
to re-use elsewhere in configuration, the same can now be achieved using
│
 
locals
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./examples/automatic-labelling-folder
╷
│
 
Warning: 
Deprecated
│
 
│
 
  with module.localhost_function.data.null_data_source.wait_for_files,
│
 
  on ../../main.tf line 42, in data "null_data_source" "wait_for_files":
│
 
  42: data "null_data_source" "wait_for_files" 
{
│
 
│
 
The null_data_source was historically used to construct intermediate values
│
 
to re-use elsewhere in configuration, the same can now be achieved using
│
 
locals
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./examples/automatic-labelling-from-localhost
╷
│
 
Warning: 
Deprecated
│
 
│
 
  with module.localhost_function.data.null_data_source.wait_for_files,
│
 
  on ../../main.tf line 42, in data "null_data_source" "wait_for_files":
│
 
  42: data "null_data_source" "wait_for_files" 
{
│
 
│
 
The null_data_source was historically used to construct intermediate values
│
 
to re-use elsewhere in configuration, the same can now be achieved using
│
 
locals
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./examples/automatic-labelling-from-repository
╷
│
 
Error: 
Unsupported argument
│
 
│
 
  on main.tf line 64, in module "repository_function":
│
 
  64:   
max_instances
 = 3000
│
 
│
 
An argument named "max_instances" is not expected here.
╵
terraform_validate ./examples/delete-vms-without-cmek
╷
│
 
Warning: 
Deprecated
│
 
│
 
  with module.localhost_function.data.null_data_source.wait_for_files,
│
 
  on ../../main.tf line 42, in data "null_data_source" "wait_for_files":
│
 
  42: data "null_data_source" "wait_for_files" 
{
│
 
│
 
The null_data_source was historically used to construct intermediate values
│
 
to re-use elsewhere in configuration, the same can now be achieved using
│
 
locals
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./examples/dynamic-files
╷
│
 
Warning: 
Deprecated
│
 
│
 
  with module.localhost_function.data.null_data_source.wait_for_files,
│
 
  on ../../main.tf line 42, in data "null_data_source" "wait_for_files":
│
 
  42: data "null_data_source" "wait_for_files" 
{
│
 
│
 
The null_data_source was historically used to construct intermediate values
│
 
to re-use elsewhere in configuration, the same can now be achieved using
│
 
locals
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./modules/event-folder-log-entry
Success!
 The configuration is valid.
terraform_validate ./modules/event-project-log-entry
Success!
 The configuration is valid.
terraform_validate ./modules/repository-function
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/automatic-labelling-folder
╷
│
 
Warning: 
Deprecated
│
 
│
 
  with module.automatic_labelling_folder.module.localhost_function.data.null_data_source.wait_for_files,
│
 
  on ../../../main.tf line 42, in data "null_data_source" "wait_for_files":
│
 
  42: data "null_data_source" "wait_for_files" 
{
│
 
│
 
The null_data_source was historically used to construct intermediate values
│
 
to re-use elsewhere in configuration, the same can now be achieved using
│
 
locals
╵
Success!
 The configuration is valid, but there were some
validation warnings as shown above.
terraform_validate ./test/setup
Success!
 The configuration is valid.
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
examples/automatic-labelling-from-repository/main.tf
--- old/examples/automatic-labelling-from-repository/main.tf
+++ new/examples/automatic-labelling-from-repository/main.tf
@@ -58,9 +58,9 @@
 module "repository_function" {
   source = "../../modules/repository-function"
 
-  description   = "Labels resource with owner information."
-  entry_point   = "labelResource"
-  runtime       = "nodejs10"
+  description = "Labels resource with owner information."
+  entry_point = "labelResource"
+  runtime     = "nodejs10"
 
   environment_variables = {
     LABEL_KEY = "principal-email"
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
examples/automatic-labelling-from-repository/main.tf
--- old/examples/automatic-labelling-from-repository/main.tf
+++ new/examples/automatic-labelling-from-repository/main.tf
@@ -58,9 +58,9 @@
 module "repository_function" {
   source = "../../modules/repository-function"
 
-  description   = "Labels resource with owner information."
-  entry_point   = "labelResource"
-  runtime       = "nodejs10"
+  description = "Labels resource with owner information."
+  entry_point = "labelResource"
+  runtime     = "nodejs10"
 
   environment_variables = {
     LABEL_KEY = "principal-email"
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by apeabody** [comment]
@bharathkkb - Here is the revised version for the CI to work around https://github.com/hashicorp/terraform-provider-google/issues/13263

File: test/setup/versions.tf [hunk 1]
```diff
@@ -15,13 +15,16 @@
...
+  required_version = ">= 0.13"
+  required_providers {
 
-provider "google-beta" {
-  version = ">= 3.53.0, < 5.0"
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 3.53.0, < 5.0"
+    }
+    google-beta = {
```
Comment (g-awmalik) [inline test/setup/versions.tf:25]: Do we really need this? `max_instances` appears to be available in GA as well.

**Review by g-awmalik — COMMENTED** [review]
LGTM, one nit on the provider.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**